sphinxcontrib/phpdomain.py

```diff
@@ -329,9 +330,10 @@ def add_target_and_index(self, name_cls, sig, signode):
             objects[fullname] = (self.env.docname, self.objtype)
 
         indextext = self.get_index_text(modname, name_cls)
-        if indextext:
-            self.indexnode['entries'].append(('single', indextext,
-                                              fullname, fullname, None))
+        if 'noindexentry' not in self.options:
+            if indextext:
```

Comment:
Couldn't these conditions be combined?

Reply:
I initially wanted to make the same change as [this one](https://github.com/sphinx-doc/sphinx/pull/10807/commits/3a4778f158a765e0c6745075778d6c971b390c7b#diff-4440aa368513769284c3d18efa39fc65c05f3f7ed3deeabbc2f2def311ae7e2b) to keep the code similar between this domain and the official ones from Sphinx, but I managed to miss a line. It should have been like this:

```python
        if 'noindexentry' not in self.options:
            indextext = self.get_index_text(modname, name_cls)
            if indextext:
                self.indexnode['entries'].append(('single', indextext,
                                                  fullname, fullname, None))
```

Would you be OK if I push such a change ?

Reply:
Yeah another commit is appreciated.

Reply:
@markstory: Done!

I also added a commit that adds `:noindex:` to the phpdomain example entries in the README. I first thought about adding `:noindexentry:` to prevent adding them to the index, then I saw in Sphinx doc that simply `:noindex:` seemed more appropriate:

> If you want to typeset an object description, without even making it available for cross-referencing, you can give the directive option flag `:noindex:` (which implies `:noindexentry:`).